crates/sncast/tests/e2e/multicall/run.rs

```diff
@@ -1,22 +1,39 @@
-use crate::helpers::constants::MULTICALL_CONFIGS_DIR;
-use crate::helpers::fixtures::default_cli_args;
+use crate::helpers::constants::{ACCOUNT_FILE_PATH, MULTICALL_CONFIGS_DIR, URL};
+use crate::helpers::fixtures::{default_cli_args, get_accounts_path};
 use crate::helpers::runner::runner;
 use std::path::Path;
+use tempfile::tempdir;
 
 #[tokio::test]
 async fn test_happy_case() {
-    let mut args = default_cli_args();
-    args.append(&mut vec!["--account", "user2"]);
-
-    let path = project_root::get_project_root().expect("failed to get project root path");
-    let path = Path::new(&path)
-        .join(MULTICALL_CONFIGS_DIR)
-        .join("deploy_invoke.toml");
-    let path_str = path.to_str().expect("failed converting path to str");
-
-    args.append(&mut vec!["multicall", "run", "--path", path_str]);
-
-    let snapbox = runner(&args);
+    let temp_dir = tempdir().expect("Unable to create temporary directory");
```

Comment:
is there any gain from using tempdirs for multicalls? I was under impression that they do not tent to change fs in any way, so maybe it is wasteful to do this?

Reply:
There were always some test cases that relied on reading the fs, even though they didn't modify it, and these tests still utilized tempdirs. 

I thought that for consistency we should then proceed and use temp dirs everywhere we interact with fs in the tests, aside from for example `--accounts-file` that is tested separately.

Reply:
reading fs is fine (unless we need salted contracts), it's the writing that may be troublesome and should use tempdirs. Creating a tempdir just to read a file (and that read does not conflict with anything) is in my opinion excessive, potentially it can slow down our tests a bit

Reply:
Ok, that makes sense. I wonder if we should then remove temp dirs in tests that had them prior to my changes, but didn't alter fs.

Reply:
Do we have such tests? It may make sense, but I would be inclined to maybe do this in different PR, if the change is big

Reply:
> Do we have such tests?

Yes, for example `main_tests::test_happy_case_from_sncast_config`

> It may make sense, but I would be inclined to maybe do this in different PR, if the change is big

That makes sense, I'm just a bit unsure about what should be our decision process for picking which tests to execute in tempdir and which not to, both for my changes and tests that already used temp dirs. How should we handle:
1. Tests which are supposed to change fs, but since we expect the error, they never do (e.g. `account::deploy::test_deploy_keystore_no_status`)
2. Same kind of tests, but specifically those that use copied contract dirs (e.g. `declare::scarb_build_fails_manifest_does_not_exist`)

Reply:
> Tests which are supposed to change fs, but since we expect the error, they never do (e.g. account::deploy::test_deploy_keystore_no_status)

But they might if test fails (so tx for some reason goes through), so this should be in tempdir imo

> Same kind of tests, but specifically those that use copied contract dirs (e.g. declare::scarb_build_fails_manifest_does_not_exist)

temdpris as well, basically the same scenario as 1. I suppose all declare test should use tempdirs, as they also build contracts/scripts. For other tests - it depends on what they actually do/can do